src/cmakeProject.ts

```diff
@@ -2659,6 +2659,13 @@ export class CMakeProject {
         // Add debug configuration from settings.
         const userConfig = this.workspaceContext.config.debugConfig;
         Object.assign(debugConfig, userConfig);
+
+        if (debugConfig.environment) {
+            for (const env of debugConfig.environment) {
+                env.value = await expandString(env.value, await this.getExpansionOptions());
```

Comment:
I recommend calling this once outside the loop and saving the value to reuse here.